secure-boot-example/README.md

```diff
@@ -1,8 +1,153 @@
+* Set the `nRPIBOOT` jumper which is labelled `Fit jumper to disable eMMC Boot' on the Compute Module 4 IO board.
+* Connect the micro USB cable to the `USB slave` port on the CM4 IO board.
+* Power cycle the CM4 IO board.
+* Connect the USB serial adapter to [GPIO 14/15](https://www.raspberrypi.com/documentation/computers/os.html#gpio-and-the-40-pin-header) on the 40-pin header.
+
+### Reset the Compute Module EEPROM
+Enable `rpiboot` mode and flash the latest EEPROM image
+```bash
+./rpiboot -d recovery
+```
+
+### Boot the example image
+Enable `rpiboot` mode and load the OS via `rpiboot` without enabling code-signing.
+```bash
+./rpiboot -d secure-boot-example
+```
+The OS should load and show activity on the boot UART and HDMI console.
+
+### Generate a signing key
+Secure-boot requires a 2048 bit RSA private key. You can either use a pre-existing
+key or generate an specific key for this example. The `KEY_FILE` environment variable
+used in the following instructions must contain the absolute path to the RSA private key in
+PEM format.
+```bash
+openssl genrsa 2048 > private.pem
+export KEY_FILE=$(pwd)/private.pem
+```
+
+### Update the EEPROM to require signed OS images
+Enable `rpiboot` mode and flash the bootloader EEPROM with updated setting enables code-signing.
+
+The `boot.conf` file sets the `SIGNED_BOOT` property `1` which instructs the bootloader to only
+load files from `boot.img` and verify this against the signature in `boot.sig`. The `update-pieeprom.sh`
+generates the signed `pieeprom.bin` image.
+
+```bash
+cd secure-boot-recovery
+# Generate the signed EEPROM image.
+../tools/update-pieeprom.sh -k "${KEY_FILE}"
+cd ..
+./rpiboot -d secure-boot-recovery
+```
+
+At this stage the signed image requirement can be reverted by flashing a default, unsigned image. 
+However, once the [OTP secure-boot flags](../secure-boot-recovery/README.md#locking-secure-boot-mode) are set then `SIGNED_BOOT` is permenantely enabled and cannot be overriden via the EEPROM config.
+
+
+### Update the signature for the example OS image
+```bash
+cd secure-boot-example
+../tools/rpi-eeprom-digest -i boot.img -o boot.sig -k "${KEY_FILE}"
+cd ..
+```
+
+### Launch the signed OS image
+Enable `rpiboot` mode and run the example OS as before. However, if the
+`boot.sig` signature does not match `boot.img` the bootloader will refuse to
+load the OS.
+
+```bash
+./rpiboot -d secure-boot-example
+```
+
+This example OS image is minimal Linux ramdisk image. Login as `root` with the empty password.
+
+### Mount the CM4 SD/EMMC after enabling secure-boot
+Now that `SIGNED_BOOT` is enabled the bootloader will only load images signed with private key generated earlier. 
+To boot the Compute Module in mass-storage mode a signed version of this code must be generated.  
+  
+**This signed-image should not be made available for download because it gives access to the EMMC as a block device**
+
+
+#### Sign the mass storage firmware image
+```bash
+cd secure-boot-msd
```

Comment:
I wonder if it's worth - near the top - explaining that the different sub folders are used OS images to boot the system via rpiboot for different purposes. As a newby it's maybe non-obvious what's going on here?

Reply:
Added a link to top level README where this is documented